src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -72,9 +72,10 @@ public T remove(int index) {
     public boolean remove(T object) {
         Node<T> newNode = new Node<>(null, object, null);
         for (int i = 0; i < size; i++) {
-            if (newNode.value == find(i).value
-                    || newNode.value != null && newNode.value.equals(find(i).value)) {
-                unlink(find(i));
+            Node<T> node = find(i);
```

Comment:
You call find method in each iteration, this is not a good solution. You need only once find node and then remove it. For example, you can write a separate method that returns node by value.